@paima/wallet

acedward opened the pull request:
# Added package for connecting web wallets
* Algorand
* Avail
* Cardano
* Evm Injected
* Evm Ethers
* (NYI) Midnight
* Mina
* Polkadot

# Pending Tasks for different PRs. Added to issue racker

- [ ] Type checking of the concise input for Wallet Methods (e.g., sendTransaction)
- [ ] Signing and Validation for other wallets than EVM
- [ ] Some injected wallets don't want to work as expected
- [ ] Implement a Local Wallet as default wallet to select
- [ ] Implement Midnight Injected (Implement midnight local?)
- [ ] Convert Ethers to Viem
- [ ] Dynamic imports for wallets
- [ ] Implement the namespace/app name in the message that are signed (?)

acedward committed:
Implemented wallet-e2e web
Added Algorand Crypto Algos
Created @paima/wallets package to import
Algorand, Avail, Cardano, EvmInjected, EvmEthers, Polkadot.

SebastienGllmt commented on packages/paima-sdk/crypto/src/chains/algorand.ts:
```diff
@@ -6,6 +6,34 @@ import {
+  makePaymentTxnWithSuggestedParamsFromObject,
+  decodeSignedTransaction,
+  type SignedTransaction,
+} from "algosdk";
```
as a reminder, we were trying to avoid static imports for these because 99% of dApps will not use Algorand at all, and it saves a lot of app startup time to not include different chains (both because of reduction in bundled file size, but also because some of these libraries initialize a bunch of WASM stuff / crypto stuff on load)

acedward replied:
Note: there are some improvements in the next PRs for the dependencies (but still not dynamic) we first want to make it working completely and bundlable

acedward committed:
EVM Wallet transactions for PaimaL2 & standard-batcher working.
Added connectors for wallets
E2E: Added DISABLE_MIDNIGHT flag for faster evm only.
Package: @e2e/wallets-ui for testing wallets
Fix for invalid utf8 encoding for postgres

acedward committed:
Added "Account Type" each type a wallet address is used: PaimaL2 / Batcher Inputs / Signing Methods
Fixed PaimaL2 data ambiguities
Updated Account Grammar
Updated e2e tests

acedward committed:
Fixes for new account grammar

acedward committed:
fix for &unlinkAddress for main account

acedward committed:
Bump to version 0.3.31
Added jsr:@paima/wallets package
Updated e2e/wallet-ui

acedward committed:
bumped to version 0.3.32
